pkgs/node/src/chain.zig

```diff
@@ -494,6 +493,108 @@ pub const BeamChain = struct {
         });
     }
 
+    /// Update block database with block, state, and slot indices
+    fn updateBlockDb(self: *Self, signedBlock: types.SignedBlockWithAttestation, blockRoot: types.Root, postState: types.BeamState, slot: types.Slot, finalizedSlot: types.Slot) !void {
+        var batch = self.db.initWriteBatch();
+        defer batch.deinit();
+
+        // Store block and state
+        batch.putBlock(database.DbBlocksNamespace, blockRoot, signedBlock);
+        batch.putState(database.DbStatesNamespace, blockRoot, postState);
+
+        // update unfinalized slot index
+        if (slot > finalizedSlot) {
+            const existing_blockroots = self.db.loadUnfinalizedSlotIndex(database.DbUnfinalizedSlotsNamespace, slot) orelse &[_]types.Root{};
+            if (existing_blockroots.len > 0) {
+                defer self.allocator.free(existing_blockroots);
+                var updated_blockroots = std.ArrayList(types.Root).init(self.allocator);
+                defer updated_blockroots.deinit();
+
+                updated_blockroots.appendSlice(existing_blockroots) catch {};
+                updated_blockroots.append(blockRoot) catch {};
+
+                batch.putUnfinalizedSlotIndex(database.DbUnfinalizedSlotsNamespace, slot, updated_blockroots.items);
+            }
+        }
+
+        // Update finalized slot indices and cleanup if finalization has advanced
+        if (finalizedSlot > self.last_emitted_finalized_slot) {
+            self.processFinalizationAdvancement(&batch, self.last_emitted_finalized_slot, finalizedSlot) catch |err| {
+                self.module_logger.err("Failed to process finalization advancement from slot {d} to {d}: {any}", .{
+                    self.last_emitted_finalized_slot,
+                    finalizedSlot,
+                    err,
+                });
+            };
+        }
+
+        self.db.commit(&batch);
+    }
+
+    /// Process finalization advancement: move canonical blocks to finalized index and cleanup unfinalized indices
+    fn processFinalizationAdvancement(self: *Self, batch: *database.Db.WriteBatch, previousFinalizedSlot: types.Slot, finalizedSlot: types.Slot) !void {
+        // 1. Fetch all newly finalized roots
+        const current_finalized = self.forkChoice.fcStore.latest_finalized.root;
+
+        const newly_finalized_roots = try self.forkChoice.getAncestorsofFinalized(self.allocator, current_finalized, previousFinalizedSlot);
```

Comment:
does this includes current_finalized? if not then current finalized also needs to be added to newly_finalized_roots?